packages/rector-generator/src/Provider/NodeTypesProvider.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\RectorGenerator\Provider;
+
+use ReflectionClass;
+use Symfony\Component\Finder\Finder;
+
+final class NodeTypesProvider
+{
+    private const PHP_PARSER_NODES_PATH = __DIR__ . '/../../../../vendor/nikic/php-parser/lib/PhpParser/Node';
+
+    private const PHP_PARSER_NAMESPACE = '\PhpParser\Node\\';
+
+    /**
+     * @var Finder
+     */
+    private $finder;
+
+    public function __construct()
+    {
+        $this->finder = new Finder();
```

Comment:
The `Finder` keeps inner state, so it should not be used as service, but created right in the place it's used